R/get_vpts_aloft.R

```diff
@@ -0,0 +1,122 @@
+#' Get VPTS data from aloftdata
+#'
+#' This function retrieves VPTS (Vertical Profile Time Series) data from the
+#' aloftdata server.
+#'
+#' - Constructs the S3 paths for the VPTS files based on the input
+#' - Performs parallel HTTP requests to fetch the vpts-csv data
+#' - Parses the response bodies with some assumptions about the column classes
+#' - Adds a column with the radar source
+#' - Overwrites the radar column with the radar_odim_code, all other values for
+#' this column are considered in error.
+#'
+#'
+#' @param radar_odim_code The radar ODIM code.
+#' @param rounded_interval The interval to fetch data for, rounded to nearest
+#'   day
+#' @param source The source of the data. One of baltrad, uva or ecog-04003.
+#' @param coverage A data.frame containing the coverage of the aloft data
+#'   repository. If not provided, it will be fetched from via the internet.
+#' @return A list of vpts data.frames
+#'
+#' @importFrom dplyr .data
+#' @importFrom lubridate %within%
+#'
+#' @examples
+#' get_vpts_aloft(
+#'   "bejab",
+#'   lubridate::interval("20240305", "20240307"),
+#'   "baltrad"
+#' )
+#'
+#' @noRd
+get_vpts_aloft <- function(radar_odim_code,
+                           rounded_interval,
+                           source,
+                           coverage = aloft_data_coverage()) {
+  # rename source argument for readability
+  selected_source <- source
+
+  # Check that only one radar is provided (string of length 1)
+  if (!rlang::is_string(radar_odim_code)) {
+    cli::cli_abort(
+      "Please provide (only one) radar as a character vector of length 1.",
+      class = "getRad_error_radar_not_single_string"
+    )
+  }
+
+  # Check that radar_odim_code is a single 5 character string
+  if (nchar(radar_odim_code) != 5) {
+    cli::cli_abort(
+      "Radar ODIM code must be a single 5 character string.",
+      class = "getRad_error_radar_odim_code_invalid"
+    )
+  }
```

Comment:
As `get_pvol` also checks for odim codes is it worth extracting this to one function?

Reply:
Separate issue for this.